src/types.d.ts

```diff
@@ -79,3 +79,26 @@ export type ApexGuruAuthResponse = {
     status: string;
 }
 
+export type ApexGuruInitialResponse = {
+    status: string;
+	requestId: string;
+	message: string;
+}
+
+export type ApexGuruQueryResponse = {
+	status: string;
+	message: string;
+	report: string;
+}
+
+export type ApexGuruProperty = {
+    name: string;
+    value: string;
+};
+
+export type ApexGuruReport = {
+	id: string;
+    type: string;
+    value: string;
+    properties: ApexGuruProperty[];
+}
```

Comment:
Why are all these types in a d.ts file? Adding a d.ts file is only needed if you are interacting with a javascript library that doesn't have its types exported for use in typescript. 

Since you are already in typescript land, the tsconfig file can generate the d.ts files automatically if you wanted to export these types to be used by other typescript modules.

Reply:
I have moved the new types from d.ts file. I will create a separate PR right after this to get rid of d.ts file altogether.